build-tools/xa-prep-tasks/Xamarin.Android.BuildTools.PrepTasks/Which.cs

```diff
@@ -30,10 +30,13 @@ static Which ()
 		{
 			var pathExt     = Environment.GetEnvironmentVariable ("PATHEXT");
 			var pathExts    = pathExt?.Split (new char [] { Path.PathSeparator }, StringSplitOptions.RemoveEmptyEntries);
-			FileExtensions  = new string [(pathExts?.Length ?? 0) + 1];
-			FileExtensions [0] = null;
+			bool isUnix     = Environment.OSVersion.Platform != PlatformID.Win32NT;
+
+			//NOTE: Unix should accept blank extensions first and Windows last
+			FileExtensions = new string [(pathExts?.Length ?? 0) + 1];
+			FileExtensions [isUnix ? 0 : FileExtensions.Length - 1] = null;
```

Comment:
The `isUnix` check isn't necessary. On Unix, `$PATHEXT` will be empty, so `FileExtensions.Length` is only 1.

Instead, just make the null entry last:

```csharp
FileExtensions  = new string [(pathExts?.Length ?? 0) + 1];
if (pathExts != null) {
    Array.Copy (pathExts, 0, FileExtensions, 0, pathExts.Length);
}
FileExtensions [FileExtensions.Length-1] = null;
```